### PR TITLE
feat(api): Add /issues/<group_id>/events/<event_id> endpoint

### LINKED ISSUE #47520: Create issue event endpoint that doesn't require project slug
The issue details page currently uses `ProjectEventDetailsEndpoint` when a specific event is requested. This endpoint requires a project slug to be sent in the URL, something that isn't known on initial pageload.

We want a new endpoint that does not require this. The `/latest` and `/oldest` endpoints work this way and can be combined into a single endpoint that can optionally accept an event_id.

### PR DESCRIPTION
**Pull request by malwilley**
Closes https://github.com/getsentry/sentry/issues/47520

This PR combines the `/issues/<group_id>/events/latest/` and `/issues/<group_id>/events/oldest/` endpoints and additionally allows you to pass an event ID.

The issues page currently uses the `/<project_slug>/events/<event_id>/` endpoint, which requires a project slug. Since the frontend doesn't know the project slug on initial load, the new `/issues/<group_id>/events/<event_id>` endpoint is preferable.

### CODE REVIEW

File: src/sentry/api/endpoints/group_events_oldest.py
Comment (malwilley) [inline src/sentry/api/endpoints/group_events_oldest.py:1]: Moved these tests into `sentry/.../test_group_event_details.py`

File: tests/sentry/api/endpoints/test_group_events_oldest.py
Comment (malwilley) [inline tests/sentry/api/endpoints/test_group_events_oldest.py:1]: Moved these tests into `sentry/.../test_group_event_details.py`

File: tests/snuba/api/endpoints/test_group_events_latest.py
Comment (malwilley) [inline tests/snuba/api/endpoints/test_group_events_latest.py:1]: Moved these tests into `snuba/.../test_group_event_details.py`

File: tests/snuba/api/endpoints/test_group_events_oldest.py
Comment (malwilley) [inline tests/snuba/api/endpoints/test_group_events_oldest.py:1]: Moved these tests into `snuba/.../test_group_event_details.py`

File: src/sentry/api/endpoints/group_event_details.py [hunk 1]
```diff
@@ -37,11 +38,20 @@ def get(self, request: Request, group: Group) -> Response:
...
+        event = None
+
+        if event_id == "latest":
+            event = group.get_latest_event_for_environments(environments)
+        elif event_id == "oldest":
+            event = group.get_oldest_event_for_environments(environments)
+        else:
+            event = eventstore.get_event_by_id(group.project.id, event_id, group_id=group.id)
+            # TODO: Remove `for_group` check once performance issues are moved to the issue platform
+            if hasattr(event, "for_group") and event.group:
```
Comment (malwilley) [inline src/sentry/api/endpoints/group_event_details.py:50]: This logic is taken from the project event details endpoint: https://github.com/getsentry/sentry/blob/6550621af105961899e897dfe278e0468d5764be/src/sentry/api/endpoints/project_event_details.py#L88-L90

**Review by scefali — APPROVED** [review]

**Review by markstory — APPROVED** [review]

**Pull request merged**